docs/conventions/index.rst

```diff
@@ -0,0 +1,141 @@
+===========
+Conventions
+===========
+
+Why Conventions?
+~~~~~~~~~~~~~~~~
+
+Zarr Conventions provide a mechanism to standardize metadata and layout of Zarr data
+in order to meet domain-specific application needs without changes to the
+core Zarr data model and specification, and without specification extensions.
+
+Conventions must fit completely within the Zarr data / metadata model of groups, arrays, and attributes thereof, requiring
+no changes or extension to the specification.
+A Zarr implementation itself should not even be aware of the existence of the convention.
+The line between a convention and an extension may be blurry in some cases.
+The key distinction lies in the implementation: the responsibility for interpreting a *convention* relies completely with downstream,
+domain-specific software, while an *extension* must be handled by the Zarr implementation itself.
+A good rule of thumb is that a user should be able to safely ignore the convention and still be able to interact with the data via the core Zarr library,
+even if some domain-specific context or functionality is missing.
+If the data are completely meaningless or unintelligible without the convention, then it should be an extension instead.
+
+Conventions can also help users switch between different storage libraries more flexibly.
+Since Zarr and HDF5 implement nearly identical data models, a single convention can be applied to both formats.
+This allows downstream software to maintain better separation of concerns between storage and domain-specific logic.
+
+Conventions are modular and composable. A single group or array can conform to multiple conventions.
+
+
+Describing Conventions
+~~~~~~~~~~~~~~~~~~~~~~
+
+Conventions Document
+--------------------
+
+Conventions are described by a *convention document*.
+TODO: say more about the structure and format of this document
+
+Explicit Conventions
+--------------------
+
+The preferred way of identifying the presence of a convention in a Zarr group or array is via the attribute `zarr_conventions`.
+This attribute must be an array of strings; each string is an identifier for the convention.
+Multiple conventions may be present.
+
+For example, a group metadata JSON document with conventions present might look like this
+
+.. code-block:: json
+
+   {
+      "zarr_format": 3,
+      "node_type": "group",
+      "attributes": {
+         "zarr_conventions": ["units-v1", "foo"],
+      }
```

Comment:
### Choice of schema-language

wanted to create a separate thread since prior one is overloaded. If the idea of "reference/contain a schema for a convention" would generally be accepted, might be worth to look into defining it in https://linkml.io/ instead of jsonschema since 1. more human readable/friendly ; 2. can be converted to jsonschema (or pydantic or ... see https://linkml.io/linkml/intro/overview.html#feature-rich-modeling-language )

Might be easier to establish such schemas.  Not yet sure if would be easier to use in some cases, so might be worthwhile accompanying with both linkml and jsonschema urls... sorry if I am adding another level of complexity right away - but wanted to establish the "target horizon" right away ;-)